packages/expo-modules-core/ios/JSI/EXJavaScriptValue.mm

```diff
@@ -9,75 +9,79 @@
 
 @implementation EXJavaScriptValue {
   __weak EXJavaScriptRuntime *_runtime;
-  std::shared_ptr<jsi::Value> _value;
+
+  /**
+   The underlying JS value of which the `JavaScriptValue` is the only owner.
+   */
+  jsi::Value _value;
 }
 
 - (nonnull instancetype)initWithRuntime:(nullable EXJavaScriptRuntime *)runtime
-                                  value:(std::shared_ptr<jsi::Value>)value
+                                  value:(jsi::Value)value
 {
   if (self = [super init]) {
     _runtime = runtime;
-    _value = value;
+    _value = std::move(value);
   }
   return self;
 }
 
-- (nonnull jsi::Value *)get
+- (jsi::Value)get
 {
-  return _value.get();
+  return jsi::Value(*[_runtime get], _value);
```

Comment:
nit: not sure if that's intentional. i think it would be faster to return
```
- (nonnull jsi::Value *)get
{
  return _value;
}
```

it should come with little overhead from creating a new jsi::Value from runtime